ThirdPartyAdapters/applovin/applovin/src/main/java/com/applovin/mediation/AppLovinNativeAdapter.java

```diff
@@ -0,0 +1,54 @@
+package com.applovin.mediation;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.util.Log;
+
+import com.applovin.sdk.AppLovinSdk;
+import com.google.android.gms.ads.AdRequest;
+import com.google.android.gms.ads.mediation.MediationNativeAdapter;
+import com.google.android.gms.ads.mediation.MediationNativeListener;
+import com.google.android.gms.ads.mediation.NativeMediationAdRequest;
+
+public class AppLovinNativeAdapter
+        implements MediationNativeAdapter {
+    protected static final String TAG = AppLovinNativeAdapter.class.getSimpleName();
+
+    public static final String KEY_EXTRA_AD_ID = "ad_id";
+    public static final String KEY_EXTRA_CAPTION_TEXT = "caption_text";
+
+    @Override
+    public void requestNativeAd(final Context context,
+                                final MediationNativeListener mediationNativeListener,
+                                final Bundle serverParameters,
+                                final NativeMediationAdRequest nativeMediationAdRequest,
+                                final Bundle mediationExtras) {
+        if (nativeMediationAdRequest.isContentAdRequested()
+                && !nativeMediationAdRequest.isAppInstallAdRequested()) {
+            Log.w(TAG, "Failed to request native ad. App install format needs to be requested");
+            mediationNativeListener
+                    .onAdFailedToLoad(AppLovinNativeAdapter.this, AdRequest.ERROR_CODE_INVALID_REQUEST);
+            return;
+        }
+
+        final AppLovinSdk sdk = AppLovinSdk.getInstance(context);
```

Comment:
I think we have a Utility method, to retrieve the AppLovinSdk. b/c the SDK key could either be passed in via `serverParameters`, or exist in the publisher's AndroidManifest. Check other adapters for example.